fcrepo-serialization/src/main/java/org/fcrepo/camel/serialization/SerializationRouter.java

```diff
@@ -0,0 +1,125 @@
+import org.fcrepo.camel.RdfNamespaces;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * A router for serializing fedora objects and binaries.
+ *
+ * @author Bethany Seeger
+ * @since 2015-09-28
+ */
+
+public class SerializationRouter extends RouteBuilder {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SerializationRouter.class);
+
+    private static final String isBinaryResourceXPath =
+        "/rdf:RDF/rdf:Description/rdf:type[@rdf:resource='" + RdfNamespaces.REPOSITORY + "Binary']";
+    /**
+     * Configure the message route workflow
+     *
+     */
+
+    public void configure() throws Exception {
+
+        final Namespaces ns = new Namespaces("rdf", RdfNamespaces.RDF);
+        ns.add("fedora", RdfNamespaces.REPOSITORY);
+
+        //boolean writeBinaries = Boolean.parseBoolean(
+         //       getContext().resolvePropertyPlaceholders("{{serialization.includeBinaries}}"));
+
+        /**
+         * A generic error handler (specific to this RouteBuilder)
+         */
+        onException(Exception.class)
+                .maximumRedeliveries("{{error.maxRedeliveries}}")
+                .log("Index Routing Error: ${routeId}");
+
+        /**
+         * Handle Serialization Events
+         */
+        from("{{input.stream}}")
+            .routeId("FcrepoSerialization")
+            .filter(not(or(header(FCREPO_IDENTIFIER).startsWith(simple("{{audit.container}}/")),
+                    header(FCREPO_IDENTIFIER).isEqualTo(simple("{{audit.container}}")))))
+            .choice()
+                .when(header(EVENT_TYPE).isEqualTo(RdfNamespaces.REPOSITORY + "NODE_REMOVED"))
+                    .multicast().to("direct:delete.metadata", "direct:delete.binary").endChoice()
+                .otherwise()
+                    .multicast().to("direct:metadata", "direct:binary");
+
+        from("{{serialization.stream}}")
+            .routeId("FcrepoReSerialization")
+            .filter(not(or(header(FCREPO_IDENTIFIER).startsWith(simple("{{audit.container}}/")),
+                    header(FCREPO_IDENTIFIER).isEqualTo(simple("{{audit.container}}")))))
+            .multicast().to("direct:metadata", "direct:binary");
+
+        from("direct:metadata")
+            .routeId("FcrepoSerializationMetadataUpdater")
+            .to("fcrepo:{{fcrepo.baseUrl}}?accept={{serialization.mimeType}}")
+            .log(INFO, LOGGER,
+                    "Serializing object ${headers[CamelFcrepoIdentifier]}")
+            .setHeader(FILE_NAME)
+                .simple("${headers[CamelFcrepoIdentifier]}.{{serialization.extension}}")
+            .log(DEBUG, LOGGER, "filename is ${headers[CamelFileName]}")
+            .to("{{serialization.descriptions}}");
+
+        from("direct:binary")
+            .routeId("FcrepoSerializationBinaryUpdater")
+            .filter().simple("{{serialization.includeBinaries}} == 'true'")
+            .to("fcrepo:{{fcrepo.baseUrl}}")
```

Comment:
I would recommend using a Prefer Header here to keep the response sizes small. That is, for a large repository, it is possible that you would have thousands or millions of ldp:contains triples in the response, but they are not needed to simply determine the rdf:type of the object (which is all you need here). Something like this would suffice:

    .to("fcrepo:{{fcrepo.baseUrl}}?preferInclude=PreferMinimalContainer")